dev-ml/cmdliner/cmdliner-1.1.1.ebuild

```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit findlib
+
+DESCRIPTION="Declarative definition of command line interfaces for OCaml"
+HOMEPAGE="http://erratique.ch/software/cmdliner"
+SRC_URI="http://erratique.ch/software/${PN}/releases/${P}.tbz"
+
+LICENSE="ISC"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+IUSE="+ocamlopt test"
+
+RDEPEND="
+	>=dev-lang/ocaml-4.08.00:=[ocamlopt?]
+	dev-ml/result:=
+	dev-ml/findlib:=
+"
+DEPEND="${RDEPEND}
+	dev-ml/topkg
+	dev-ml/ocamlbuild"
+
+RESTRICT="!test? ( test )"
+
+src_compile() {
+	ocaml pkg/pkg.ml build \
+		--tests $(usex test true false) \
+		|| die
+}
+
+src_test() {
+	ocaml pkg/pkg.ml test || die
```

Comment:
This isn't the correct build command - see `Makefile` in the sources, you want
```sh
ocaml build.ml cma
ocaml build.ml cmxa
ocaml build.ml cmxs
```
(not sure if they can be combined)

Reply:
@Alessandro-Barbieri actually delegating this work to makefile would be better imo.

consider:
```bash
src_compile() {
    emake build-byte

    if use ocamlopt ; then
        emake build-native build-native-dynlink
    fi
}
```